arch/xtensa/soc/intel_s1000/linker.ld

```diff
@@ -0,0 +1,537 @@
+  srom1_seg :                         	org = 0xBEFE0300, len = 0x100
+  srom2_seg :                         	org = 0xBEFE0400, len = 0x180
+  srom4_seg :                         	org = 0xBEFE0800, len = 0x178
+  srom5_seg :                         	org = 0xBEFE0978, len = 0x8
+  srom6_seg :                         	org = 0xBEFE0980, len = 0x38
+  srom7_seg :                         	org = 0xBEFE09B8, len = 0x8
+  srom8_seg :                         	org = 0xBEFE09C0, len = 0x38
+  srom9_seg :                         	org = 0xBEFE09F8, len = 0x8
+  srom10_seg :                        	org = 0xBEFE0A00, len = 0x38
+  srom11_seg :                        	org = 0xBEFE0A38, len = 0x8
+  srom12_seg :                        	org = 0xBEFE0A40, len = 0x38
+  srom13_seg :                        	org = 0xBEFE0A78, len = 0x8
+  srom14_seg :                        	org = 0xBEFE0A80, len = 0x38
+  srom15_seg :                        	org = 0xBEFE0AB8, len = 0x8
+  srom16_seg :                        	org = 0xBEFE0AC0, len = 0x38
+  srom17_seg :                        	org = 0xBEFE0AF8, len = 0x8
+  srom18_seg :                        	org = 0xBEFE0B00, len = 0x38
+  srom19_seg :                        	org = 0xBEFE0B38, len = 0x8
+  srom20_seg :                        	org = 0xBEFE0B40, len = 0x38
+  srom21_seg :                        	org = 0xBEFE0B78, len = 0x48
+  srom22_seg :                        	org = 0xBEFE0BC0, len = 0x40
+  srom23_seg :                        	org = 0xBEFE0C00, len = 0x1D400
+#ifdef CONFIG_GEN_ISR_TABLES
+  IDT_LIST :                            org = 0xBEFFE000, len = 0x2000
+#endif
+}
+
+PHDRS
+{
+  sram0_phdr PT_LOAD;
+  sram0_bss_phdr PT_LOAD;
+  srom0_phdr PT_LOAD;
+  srom1_phdr PT_LOAD;
+  srom2_phdr PT_LOAD;
+  srom3_phdr PT_LOAD;
+  srom4_phdr PT_LOAD;
+  srom5_phdr PT_LOAD;
+  srom6_phdr PT_LOAD;
+  srom7_phdr PT_LOAD;
+  srom8_phdr PT_LOAD;
+  srom9_phdr PT_LOAD;
+  srom10_phdr PT_LOAD;
+  srom11_phdr PT_LOAD;
+  srom12_phdr PT_LOAD;
+  srom13_phdr PT_LOAD;
+  srom14_phdr PT_LOAD;
+  srom15_phdr PT_LOAD;
+  srom16_phdr PT_LOAD;
+  srom17_phdr PT_LOAD;
+  srom18_phdr PT_LOAD;
+  srom19_phdr PT_LOAD;
+  srom20_phdr PT_LOAD;
+  srom21_phdr PT_LOAD;
+  srom22_phdr PT_LOAD;
+  srom23_phdr PT_LOAD;
+}
+
+
+/*  Default entry point:  */
+ENTRY(_ResetVector)
+
+/*  Memory boundary addresses:  */
+_memmap_mem_sram_start = 0x80000000;
+_memmap_mem_sram_end   = 0xbefe0000;
+_memmap_mem_srom_start = 0xbefe0000;
+_memmap_mem_srom_end   = 0xbf000000;
+
+/*  Memory segment boundary addresses:  */
+_memmap_seg_sram0_start = 0x80000000;
+_memmap_seg_sram0_max   = 0xbefe0000;
+_memmap_seg_srom0_start = 0xbefe0000;
+_memmap_seg_srom0_max   = 0xbefe0300;
+_memmap_seg_srom1_start = 0xbefe0300;
+_memmap_seg_srom1_max   = 0xbefe0400;
+_memmap_seg_srom2_start = 0xbefe0400;
+_memmap_seg_srom2_max   = 0xbefe0580;
+_memmap_seg_srom4_start = 0xbefe0800;
+_memmap_seg_srom4_max   = 0xbefe0978;
```

Comment:
Sorry, who is using all these symbols?  This is a staggering amount of typing for what AFAICT is a bunch of untested numbers copied out of a book somewhere.  That sounds dangerous.  Do we really need all this segment boundaries in the linker script if we aren't putting anything there?